api/envoy/api/v2/core/protocol.proto

```diff
@@ -19,6 +19,11 @@ option (udpa.annotations.file_migrate).move_to_package = "envoy.config.core.v3al
 message TcpProtocolOptions {
 }
 
+message UpstreamHttpProtocolOptions {
+  // Set transport socket SNI for new upstream connections based on HTTP host/authority header.
```

Comment:
```suggestion
  // Set transport socket SNI for new upstream connections based on the downstream HTTP host/authority header, 
  // as seen by the router filter
```

Please also hyperlink to SNI if possible (wikipedia) and ref link to the router filter documentation for the "router filter" text.